calculation/sha_calc/sha_calc/directivity/bea20/scripts/mc_nhyp_runs.py

```diff
@@ -0,0 +1,154 @@
+import time
+import multiprocessing as mp
+from pathlib import Path
+import argparse
+
+import numpy as np
+
+import sha_calc
+from sha_calc.directivity.bea20.validation.plots import plot_fdi
+import common
+
+
+def perform_mp_directivity(
+    fault_name,
+    hypo_along_strike,
+    hypo_down_dip,
+    method,
+    repeats,
+    period,
+    grid_space,
+    nhm_dict,
+    output_dir,
+):
+    nhyp = hypo_along_strike * hypo_down_dip
+    print(f"Computing for {fault_name} {nhyp}")
+
+    fault, site_coords, planes, lon_lat_depth, x, y = common.load_fault_info(
+        fault_name, nhm_dict, grid_space
+    )
+
+    total_fd = np.zeros((repeats, len(site_coords), 1))
+    total_fd_array = np.zeros((repeats, nhyp, len(site_coords), 1))
+
+    for i in range(repeats):
+        fdi, fdi_array, phi_red = sha_calc.bea20.compute_fault_directivity(
+            lon_lat_depth,
+            planes,
+            site_coords,
+            hypo_along_strike,
+            hypo_down_dip,
+            fault.mw,
+            fault.rake,
+            periods=[period],
+            method=method,
+        )
+
+        total_fd[i] = fdi
+        total_fd_array[i] = fdi_array
+
+    fdi_average = np.mean(total_fd, axis=0)
+    fdi_average = fdi_average.reshape((grid_space, grid_space))
+
+    title = f"{fault_name} Length={fault.length} Dip={fault.dip} Rake={fault.rake}"
+    plot_fdi(
+        x,
+        y,
+        fdi_average,
+        lon_lat_depth,
+        Path(f"{output_dir}/{fault_name}_{nhyp}.png"),
+        title,
+    )
+    np.save(
+        f"{output_dir}/{fault_name}_{nhyp}_fd_mc_hypo_array.npy",
+        np.exp(total_fd_array),
+    )
+    np.save(
+        f"{output_dir}/{fault_name}_{nhyp}_fd_mc.npy",
+        np.exp(total_fd),
+    )
+    np.save(
+        f"{output_dir}/{fault_name}_{nhyp}_fd_average.npy",
+        np.exp(fdi_average),
+    )
+
+
+def parse_args():
+    nhm_dict, faults, im, grid_space, nhyps = common.default_variables()
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("output_dir")
+    parser.add_argument(
+        "--faults",
+        default=faults,
+        nargs="+",
+        help="Which faults to calculate for",
```

Comment:
I'd remove the question starter. I.E List of faults to calculate.

"list of n hypocentres along strike"
etc